mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageBatcher.java

```diff
@@ -48,11 +50,11 @@ public int getBatchSize() {
 
     public List<MessageRange> batchMessages(MessageRange set, BatchedOperation batchedOperation) throws MailboxException {
         if (batchSize > 0) {
-            List<MessageRange> movedRanges = new ArrayList<>();
-            for (MessageRange messageRange : set.split(batchSize)) {
-                movedRanges.addAll(batchedOperation.execute(messageRange));
-            }
-            return movedRanges;
+            return set.split(batchSize)
+                .stream()
+                .flatMap(Throwing.function(range -> batchedOperation.execute(range)
+                                                                    .stream()))
+                .collect(ImmutableList.toImmutableList());
```

Comment:
you should still use Guavate, ImmutableList is annotated as @Beta and generates a warning in Intellij

Reply:
are you sure com.google.common.collect.ImmutableList is issuing a warning?

Reply:
```
$ git grep ImmutableList.toImmutableList | wc -l
26
```

Reply:
can we use CollectCollectors.toImmutableList() instead?

Reply:
CollectCollectors.toImmutableList() is used in ImmutableList.toImmutableList() anyway :D
```java
  @Beta
  public static <E> Collector<E, ?, ImmutableList<E>> toImmutableList() {
    return CollectCollectors.toImmutableList();
  }
```

I think leaving `@Beta` annotation at that method is a mistake from developers of guava, [See](https://github.com/google/guava/issues/3251). And the explanation from the guy raised that issue is pretty rational to me.

Reply:
![screenshot from 2018-11-29 17-52-51](https://user-images.githubusercontent.com/708318/49238029-db85f700-f3ff-11e8-9268-efc061535f49.png)

Reply:
As far as I understand, IDEA shows warning because of `@Beta` annotation from Guava. [From](https://github.com/google/guava/wiki/PhilosophyExplained#beta-apis), I think we earned guarantee from Guava for `@Beta` annotation:
> That said, @Beta APIs are fully tested and supported, and treated with all the care and affection that the rest of Guava receives.

If this method will be removed in the future, we can easily replace it by guavate:
> The biggest connotation of the @Beta annotation is that annotated classes or methods are subject to change. They can be modified in any way, or even removed, at any time. If your code is a library itself (i.e. it is used on the CLASSPATH of users outside your own control), you should not use beta APIs, unless you repackage them (e.g. using ProGuard).

`ImmutableList.toImmutableList()` is introduced since guava version 21 till now - guava version 27 and still carries `@Beta`. The possible reason might be no one raised that removal to guava before.

> All this said, @Beta features tend to remain relatively stable. If we decide to delete a @Beta feature, we will typically deprecate it for one release before deleting it.
> On the other hand, if you want something taken out of @Beta, file an issue. We generally promote features out of @Beta only when it's specifically requested, so if you don't ask, it won't happen.

So I trust `@Beta`

Reply:
I don't share your opinion: whatever what people say or write, if they don't care to remove @beta then it's beta and we already have a library we use for a long time that do just that: collect to a ImmutableList. Why even bother discussing this solution that is inferior ? I mean, if we never changed from Guavate to Guava, we would just be happy, change for the sake of change is probably useless.